sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -52,8 +52,20 @@ public KeyVaultCredentialPolicy(TokenCredential credential) {
             return Mono.error(new RuntimeException("Token credentials require a URL using the HTTPS protocol scheme"));
         }
         return next.clone().process()
-            // Ignore body
-            .doOnNext(HttpResponse::close)
+            .doOnNext(httpResponse -> {
+                // KV follows challenge based auth. Currently every service
+                // call hit the endpoint for challenge and then resend the
+                // request with token. The challenge response body is not
+                // consumed, not draining/closing the body will result in leak.
+                // Ref: https://github.com/Azure/azure-sdk-for-java/issues/7934
+                try {
+                    httpResponse.getBody().subscribe().dispose();
+                } catch (Throwable ignored) {
```

Comment:
Should this throwable be logged as warning?

Reply:
yes, we should, will update the PR. It's not throwing now but I put it as a safeguard to ensure this version of key vault lib works with the future version of reactor-netty HTTP client when it gets addressed in the core level.